### PR TITLE
Verified list of LLMs working with OpenHands and some LLM doc changes

### PR DESCRIPTION
**Pull request by mamoodi**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Added verified list of LLMs working with OpenHands.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

For better transparency and clarity, rewriting the LLM backend docs a little bit to make it clear which ones we know for sure are working with OpenHands.

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by mamoodi** [comment]
What should we do with the current ollama guide?

File: docs/modules/usage/llms/llms.md [hunk 1]
```diff
@@ -4,16 +4,26 @@ sidebar_position: 2
 
 # 🤖 LLM Backends
 
-OpenHands can connect to many LLMs. However, the recommended models to use are GPT-4 and Claude 3.5.
+OpenHands can connect to any LLM supported by LiteLLM. However, it requires a powerful model to work.
+The following are verified by the community to work with OpenHands:
+* GPT4
+* Claude 3.5
```
Comment (mamoodi) [inline docs/modules/usage/llms/llms.md:10]: Any I missed? Just went with the "Verified" list in the UI

Comment (tobitege) [inline docs/modules/usage/llms/llms.md:10]: Should be more specific: Claude 3.5-Sonnet.

Comment (tobitege) [reply docs/modules/usage/llms/llms.md:10]: I've used below as well and see them as usable:
- llama-3.1-405b / hermes-3-llama-3.1-405b
- gemini-pro-1.5 /-flash
- wizardlm-2-8x22b

Comment (mamoodi) [reply docs/modules/usage/llms/llms.md:10]: The first one I thought was not feasible with OpenHands.

For the other two, is it:
* gemini-1.5-pro / gemini-1.5-flash
* wizardlm-2-8x22b

?

Comment (tobitege) [reply docs/modules/usage/llms/llms.md:10]: Oops... my post was for the general list, not the ollama one, sorry 😬

**Review by tobitege — APPROVED** [review]
LGTM, thanks! 🍰

**Pull request merged**